packages/terra-table/src/Table.jsx

```diff
@@ -342,14 +413,16 @@ function Table(props) {
                 hasRowSelection={hasSelectableRows}
                 displayedColumns={displayedColumns}
                 rowHeaderIndex={rowHeaderIndex}
-                onCellSelect={isGridContext ? handleCellSelection : undefined}
+                onCellSelect={isGridContext || hasSelectableRows ? handleCellSelection : undefined}
```

Comment:
I might be missing why this condition was added here. Was this added as part of allowing the row selection cell checkbox to be selectable?

Reply:
Previously the onCellSelect, which lets us know that a cell was "selected", would only trigger when in the grid context.  If we are in row selection mode, hasSelectableRows prop equal to true, then we need to also handle the cell selection.